File: tests/fixtures/python/tests/test_checksum.py
from ciwf_fixture import checksum


def test_checksum_is_stable() -> None:
    assert checksum("ci") == 204
    assert checksum("") == 0


def test_checksum_stays_in_one_byte() -> None:
    for sample in ("", "a", "the quick brown fox", "ü" * 40):
        assert 0 <= checksum(sample) <= 255

File: tests/fixtures/r/tests/testthat/test-checksum.R
test_that("checksum is stable", {
  expect_equal(checksum("ci"), 204L)
  expect_equal(checksum(""), 0L)
})

test_that("checksum stays in one byte", {
  for (sample in c("", "a", "the quick brown fox")) {
    expect_gte(checksum(sample), 0L)
    expect_lte(checksum(sample), 255L)
  }
})
